src/cpp/desktop/DesktopMainWindow.cpp

```diff
@@ -230,11 +190,16 @@ void MainWindow::quit()
 
 void MainWindow::invokeCommand(QString commandId)
 {
-   QString command =
-         QString::fromUtf8("window.desktopHooks.invokeCommand('") +
-         commandId +
-         QString::fromUtf8("');");
-
+#ifdef Q_OS_MAC
+   QString fmt = QStringLiteral(R"EOF(
+      var wnd = window.$RStudio.last_focused_window || window;
```

Comment:
Is it possible for `window.$RStudio` to be null?

Reply:
It shouldn't (this is always set on the main window) but I added a fallback just in case it is missing for some reason.